src/blockindices.jl

```diff
@@ -288,6 +288,16 @@ _indices(B) = B
 
 Block(bs::BlockSlice{<:BlockIndexRange}) = Block(bs.block)
 
+struct BlockInds{BB,T<:Integer,INDS<:AbstractVector{T}} <: AbstractVector{T}
```

Comment:
Can you add a docstring so I know what this is meant to do?

Reply:
Can do. I was holding off doing "final touches" like that to get your reaction if this is something you want in the first place.

Reply:
I can't react to something I don't understand 😅

Reply:
I've added a docstring along with some tests, let me know if that helps to clarify the purpose of this object. I changed the name to `BlockedSlice` but I'm not so happy with that and again I'm open to suggestions.